src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellContentFragment.cs

```diff
@@ -173,7 +173,7 @@ void Destroy()
 			// to avoid the navigation `TaskCompletionSource` to be stuck forever.
 			AnimationFinished?.Invoke(this, EventArgs.Empty);
 
-			((IShellController)_shellContext.Shell).RemoveAppearanceObserver(this);
+			(_shellContext?.Shell as IShellController)?.RemoveAppearanceObserver(this);
```

Comment:
Can we also add a check on this destroy method so that it only runs once?

It's fine to leave this code but I just worry that this destroy method is being called twice which might just manifest as a different problem down the road

Reply:
@postalservice14 

Could you include a variable:

`bool _destroyed;`

and update the Destroy method like:

```
void Destroy()
 {
 	if (_destroyed)
 		return;

 	_destroyed = true;
          ...
```

In that way we include a guard check to ensure the Destroy method only runs once, similar to the existing `_disposed` pattern. This prevents any potential issues if the method is called from both `OnDestroy()` and `Dispose()`.